File: src/zugferd_2_3_2/enum1001.rs
#![allow(non_camel_case_types)]

#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Hash)]
pub enum Enum1001 {
    /// Request for payment
    RequestForPayment,
    /// Debit note related to goods or services
    DebitNoteRelatedToGoodsOrServices,
    /// Credit note related to goods or services
    CreditNoteRelatedToGoodsOrServices,
    /// Metered services invoice
    MeteredServicesInvoice,
    /// Credit note related to financial adjustments
    CreditNoteRelatedToFinancialAdjustments,
    /// Debit note related to financial adjustments
    DebitNoteRelatedToFinancialAdjustments,
    /// Tax notification
    TaxNotification,
    /// Invoicing data sheet
    InvoicingDataSheet,
    /// Direct payment valuation
    DirectPaymentValuation,
    /// Provisional payment valuation
    ProvisionalPaymentValuation,
    /// Payment valuation
    PaymentValuation,
    /// Interim application for payment
    InterimApplicationForPayment,
    /// Final payment request based on completion of work
    FinalPaymentRequestBasedOnCompletionWork,
    /// Payment request for completed units
    PaymentRequestForCompletedUnits,
    /// Self billed credit note
    SelfBilledCreditNote,
    /// Consolidated credit note - goods and services
    ConsolidatedCreditNoteGoodsAndServices,
    /// Price variation invoice
    PriceVariationInvoice,
    /// Credit note for price variation
    CreditNoteForPriceVariation,
    /// Delcredere credit note
    DelcredereCreditNote,
    /// Proforma invoice
    ProformaInvoice,
    /// Partial invoice
    PartialInvoice,
    /// Commercial invoice which includes a packing list
    CommercialInvoiceWhichIncludesAPackingList,
    /// Commercial invoice
    CommercialInvoice,
    /// Credit note
    CreditNote,
    /// Commission note
    CommissionNote,
    /// Debit note
    DebitNote,
    /// Corrected invoice
    CorrectedInvoice,
    /// Consolidated invoice
    ConsolidatedInvoice,
    /// Prepayment invoice
    PrepaymentInvoice,
    /// Hire invoice
    HireInvoice,
    /// Tax invoice
    TaxInvoice,
    /// Self-billed invoice
    SelfBilledInvoice,
    /// Delcredere invoice
    DelcredereInvoice,
    /// Factored invoice
    FactoredInvoice,
    /// Lease invoice
    LeaseInvoice,
    /// Consignment invoice
    ConsignmentInvoice,
    /// Factored credit note
    FactoredCreditNote,
    /// Optical Character Reading (OCR) payment credit note
    OpticalCharacterReadingOcrPaymentCreditNote,
    /// Debit advice
    DebitAdvice,
    /// Reversal of debit
    ReversalDebit,
    /// Reversal of credit
    ReversalCredit,
    /// Self billed debit note
    SelfBilledDebitNote,
    /// Forwarder's credit note
    ForwardersCreditNote,
    /// Forwarder's invoice discrepancy report
    ForwardersInvoiceDiscrepancyReport,
    /// Insurer's invoice
    InsurersInvoice,
    /// Forwarder's invoice
    ForwardersInvoice,
    /// Port charges documents
    PortChargesDocuments,
    /// Invoice information for accounting purposes
    InvoiceInformationForAccountingPurposes,
    /// Freight invoice
    FreightInvoice,
    /// Claim notification
    ClaimNotification,
    /// Consular invoice
    ConsularInvoice,
    /// Partial construction invoice
    PartialConstructionInvoice,
    /// Partial final construction invoice
    PartialFinalConstructionInvoice,
    /// Final construction invoice
    FinalConstructionInvoice,
    /// Customs invoice
    CustomsInvoice,
}

impl crate::Code for Enum1001 {
    fn code(&self) -> &str {
        match self {
            Enum1001::RequestForPayment => "71",
            Enum1001::DebitNoteRelatedToGoodsOrServices => "80",
            Enum1001::CreditNoteRelatedToGoodsOrServices => "81",
            Enum1001::MeteredServicesInvoice => "82",
            Enum1001::CreditNoteRelatedToFinancialAdjustments => "83",
            Enum1001::DebitNoteRelatedToFinancialAdjustments => "84",
            Enum1001::TaxNotification => "102",
            Enum1001::InvoicingDataSheet => "130",
            Enum1001::DirectPaymentValuation => "202",
            Enum1001::ProvisionalPaymentValuation => "203",
            Enum1001::PaymentValuation => "204",
            Enum1001::InterimApplicationForPayment => "211",
            Enum1001::FinalPaymentRequestBasedOnCompletionWork => "218",
            Enum1001::PaymentRequestForCompletedUnits => "219",
            Enum1001::SelfBilledCreditNote => "261",
            Enum1001::ConsolidatedCreditNoteGoodsAndServices => "262",
            Enum1001::PriceVariationInvoice => "295",
            Enum1001::CreditNoteForPriceVariation => "296",
            Enum1001::DelcredereCreditNote => "308",
            Enum1001::ProformaInvoice => "325",
            Enum1001::PartialInvoice => "326",
            Enum1001::CommercialInvoiceWhichIncludesAPackingList => "331",
            Enum1001::CommercialInvoice => "380",
            Enum1001::CreditNote => "381",
            Enum1001::CommissionNote => "382",
            Enum1001::DebitNote => "383",
            Enum1001::CorrectedInvoice => "384",
            Enum1001::ConsolidatedInvoice => "385",
            Enum1001::PrepaymentInvoice => "386",
            Enum1001::HireInvoice => "387",
            Enum1001::TaxInvoice => "388",
            Enum1001::SelfBilledInvoice => "389",
            Enum1001::DelcredereInvoice => "390",
            Enum1001::FactoredInvoice => "393",
            Enum1001::LeaseInvoice => "394",
            Enum1001::ConsignmentInvoice => "395",
            Enum1001::FactoredCreditNote => "396",
            Enum1001::OpticalCharacterReadingOcrPaymentCreditNote => "420",
            Enum1001::DebitAdvice => "456",
            Enum1001::ReversalDebit => "457",
            Enum1001::ReversalCredit => "458",
            Enum1001::SelfBilledDebitNote => "527",
            Enum1001::ForwardersCreditNote => "532",
            Enum1001::ForwardersInvoiceDiscrepancyReport => "553",
            Enum1001::InsurersInvoice => "575",
            Enum1001::ForwardersInvoice => "623",
            Enum1001::PortChargesDocuments => "633",
            Enum1001::InvoiceInformationForAccountingPurposes => "751",
            Enum1001::FreightInvoice => "780",
            Enum1001::ClaimNotification => "817",
            Enum1001::ConsularInvoice => "870",
            Enum1001::PartialConstructionInvoice => "875",
            Enum1001::PartialFinalConstructionInvoice => "876",
            Enum1001::FinalConstructionInvoice => "877",
            Enum1001::CustomsInvoice => "935",
        }
    }
}

impl crate::Description for Enum1001 {
    fn description(&self) -> &str {
        match self {
            Enum1001::RequestForPayment => "Request for payment",
            Enum1001::DebitNoteRelatedToGoodsOrServices => {
                "Debit note related to goods or services"
            }
            Enum1001::CreditNoteRelatedToGoodsOrServices => {
                "Credit note related to goods or services"
            }
            Enum1001::MeteredServicesInvoice => "Metered services invoice",
            Enum1001::CreditNoteRelatedToFinancialAdjustments => {
                "Credit note related to financial adjustments"
            }
            Enum1001::DebitNoteRelatedToFinancialAdjustments => {
                "Debit note related to financial adjustments"
            }
            Enum1001::TaxNotification => "Tax notification",
            Enum1001::InvoicingDataSheet => "Invoicing data sheet",
            Enum1001::DirectPaymentValuation => "Direct payment valuation",
            Enum1001::ProvisionalPaymentValuation => "Provisional payment valuation",
            Enum1001::PaymentValuation => "Payment valuation",
            Enum1001::InterimApplicationForPayment => "Interim application for payment",
            Enum1001::FinalPaymentRequestBasedOnCompletionWork => {
                "Final payment request based on completion of work"
            }
            Enum1001::PaymentRequestForCompletedUnits => "Payment request for completed units",
            Enum1001::SelfBilledCreditNote => "Self billed credit note",
            Enum1001::ConsolidatedCreditNoteGoodsAndServices => {
                "Consolidated credit note - goods and services"
            }
            Enum1001::PriceVariationInvoice => "Price variation invoice",
            Enum1001::CreditNoteForPriceVariation => "Credit note for price variation",
            Enum1001::DelcredereCreditNote => "Delcredere credit note",
            Enum1001::ProformaInvoice => "Proforma invoice",
            Enum1001::PartialInvoice => "Partial invoice",
            Enum1001::CommercialInvoiceWhichIncludesAPackingList => {
                "Commercial invoice which includes a packing list"
            }
            Enum1001::CommercialInvoice => "Commercial invoice",
            Enum1001::CreditNote => "Credit note",
            Enum1001::CommissionNote => "Commission note",
            Enum1001::DebitNote => "Debit note",
            Enum1001::CorrectedInvoice => "Corrected invoice",
            Enum1001::ConsolidatedInvoice => "Consolidated invoice",
            Enum1001::PrepaymentInvoice => "Prepayment invoice",
            Enum1001::HireInvoice => "Hire invoice",
            Enum1001::TaxInvoice => "Tax invoice",
            Enum1001::SelfBilledInvoice => "Self-billed invoice",
            Enum1001::DelcredereInvoice => "Delcredere invoice",
            Enum1001::FactoredInvoice => "Factored invoice",
            Enum1001::LeaseInvoice => "Lease invoice",
            Enum1001::ConsignmentInvoice => "Consignment invoice",
            Enum1001::FactoredCreditNote => "Factored credit note",
            Enum1001::OpticalCharacterReadingOcrPaymentCreditNote => {
                "Optical Character Reading (OCR) payment credit note"
            }
            Enum1001::DebitAdvice => "Debit advice",
            Enum1001::ReversalDebit => "Reversal of debit",
            Enum1001::ReversalCredit => "Reversal of credit",
            Enum1001::SelfBilledDebitNote => "Self billed debit note",
            Enum1001::ForwardersCreditNote => "Forwarder's credit note",
            Enum1001::ForwardersInvoiceDiscrepancyReport => {
                "Forwarder's invoice discrepancy report"
            }
            Enum1001::InsurersInvoice => "Insurer's invoice",
            Enum1001::ForwardersInvoice => "Forwarder's invoice",
            Enum1001::PortChargesDocuments => "Port charges documents",
            Enum1001::InvoiceInformationForAccountingPurposes => {
                "Invoice information for accounting purposes"
            }
            Enum1001::FreightInvoice => "Freight invoice",
            Enum1001::ClaimNotification => "Claim notification",
            Enum1001::ConsularInvoice => "Consular invoice",
            Enum1001::PartialConstructionInvoice => "Partial construction invoice",
            Enum1001::PartialFinalConstructionInvoice => "Partial final construction invoice",
            Enum1001::FinalConstructionInvoice => "Final construction invoice",
            Enum1001::CustomsInvoice => "Customs invoice",
        }
    }
}

impl crate::FromCode for Enum1001 {
    fn from_code(code: &str) -> Option<Self>
    where
        Self: Sized,
    {
        match code {
            "71" => Some(Enum1001::RequestForPayment),
            "80" => Some(Enum1001::DebitNoteRelatedToGoodsOrServices),
            "81" => Some(Enum1001::CreditNoteRelatedToGoodsOrServices),
            "82" => Some(Enum1001::MeteredServicesInvoice),
            "83" => Some(Enum1001::CreditNoteRelatedToFinancialAdjustments),
            "84" => Some(Enum1001::DebitNoteRelatedToFinancialAdjustments),
            "102" => Some(Enum1001::TaxNotification),
            "130" => Some(Enum1001::InvoicingDataSheet),
            "202" => Some(Enum1001::DirectPaymentValuation),
            "203" => Some(Enum1001::ProvisionalPaymentValuation),
            "204" => Some(Enum1001::PaymentValuation),
            "211" => Some(Enum1001::InterimApplicationForPayment),
            "218" => Some(Enum1001::FinalPaymentRequestBasedOnCompletionWork),
            "219" => Some(Enum1001::PaymentRequestForCompletedUnits),
            "261" => Some(Enum1001::SelfBilledCreditNote),
            "262" => Some(Enum1001::ConsolidatedCreditNoteGoodsAndServices),
            "295" => Some(Enum1001::PriceVariationInvoice),
            "296" => Some(Enum1001::CreditNoteForPriceVariation),
            "308" => Some(Enum1001::DelcredereCreditNote),
            "325" => Some(Enum1001::ProformaInvoice),
            "326" => Some(Enum1001::PartialInvoice),
            "331" => Some(Enum1001::CommercialInvoiceWhichIncludesAPackingList),
            "380" => Some(Enum1001::CommercialInvoice),
            "381" => Some(Enum1001::CreditNote),
            "382" => Some(Enum1001::CommissionNote),
            "383" => Some(Enum1001::DebitNote),
            "384" => Some(Enum1001::CorrectedInvoice),
            "385" => Some(Enum1001::ConsolidatedInvoice),
            "386" => Some(Enum1001::PrepaymentInvoice),
            "387" => Some(Enum1001::HireInvoice),
            "388" => Some(Enum1001::TaxInvoice),
            "389" => Some(Enum1001::SelfBilledInvoice),
            "390" => Some(Enum1001::DelcredereInvoice),
            "393" => Some(Enum1001::FactoredInvoice),
            "394" => Some(Enum1001::LeaseInvoice),
            "395" => Some(Enum1001::ConsignmentInvoice),
            "396" => Some(Enum1001::FactoredCreditNote),
            "420" => Some(Enum1001::OpticalCharacterReadingOcrPaymentCreditNote),
            "456" => Some(Enum1001::DebitAdvice),
            "457" => Some(Enum1001::ReversalDebit),
            "458" => Some(Enum1001::ReversalCredit),
            "527" => Some(Enum1001::SelfBilledDebitNote),
            "532" => Some(Enum1001::ForwardersCreditNote),
            "553" => Some(Enum1001::ForwardersInvoiceDiscrepancyReport),
            "575" => Some(Enum1001::InsurersInvoice),
            "623" => Some(Enum1001::ForwardersInvoice),
            "633" => Some(Enum1001::PortChargesDocuments),
            "751" => Some(Enum1001::InvoiceInformationForAccountingPurposes),
            "780" => Some(Enum1001::FreightInvoice),
            "817" => Some(Enum1001::ClaimNotification),
            "870" => Some(Enum1001::ConsularInvoice),
            "875" => Some(Enum1001::PartialConstructionInvoice),
            "876" => Some(Enum1001::PartialFinalConstructionInvoice),
            "877" => Some(Enum1001::FinalConstructionInvoice),
            "935" => Some(Enum1001::CustomsInvoice),
            _ => None,
        }
    }
}

File: ts/src/zugferd_2_4/enum1001.ts
export enum Enum1001 {
  /**
   * Request for payment
   *
   * Invoice
   */
  RequestForPayment = "71",
  /**
   * Debit note related to goods or services
   *
   * Invoice
   */
  DebitNoteRelatedToGoodsOrServices = "80",
  /**
   * Credit note related to goods or services
   *
   * Credit Note
   */
  CreditNoteRelatedToGoodsOrServices = "81",
  /**
   * Metered services invoice
   *
   * Invoice
   */
  MeteredServicesInvoice = "82",
  /**
   * Credit note related to financial adjustments
   *
   * Credit Note
   */
  CreditNoteRelatedToFinancialAdjustments = "83",
  /**
   * Debit note related to financial adjustments
   *
   * Invoice
   */
  DebitNoteRelatedToFinancialAdjustments = "84",
  /**
   * Tax notification
   *
   * Invoice
   */
  TaxNotification = "102",
  /**
   * Invoicing data sheet
   *
   * Invoice
   */
  InvoicingDataSheet = "130",
  /**
   * Direct payment valuation
   *
   * Invoice
   */
  DirectPaymentValuation = "202",
  /**
   * Provisional payment valuation
   *
   * Invoice
   */
  ProvisionalPaymentValuation = "203",
  /**
   * Payment valuation
   *
   * Invoice
   */
  PaymentValuation = "204",
  /**
   * Interim application for payment
   *
   * Invoice
   */
  InterimApplicationForPayment = "211",
  /**
   * Final payment request based on completion of work
   *
   * Invoice
   */
  FinalPaymentRequestBasedOnCompletionWork = "218",
  /**
   * Payment request for completed units
   *
   * Invoice
   */
  PaymentRequestForCompletedUnits = "219",
  /**
   * Self billed credit note
   *
   * Credit Note
   */
  SelfBilledCreditNote = "261",
  /**
   * Consolidated credit note - goods and services
   *
   * Credit Note
   */
  ConsolidatedCreditNoteGoodsAndServices = "262",
  /**
   * Price variation invoice
   *
   * Invoice
   */
  PriceVariationInvoice = "295",
  /**
   * Credit note for price variation
   *
   * Credit Note
   */
  CreditNoteForPriceVariation = "296",
  /**
   * Delcredere credit note
   *
   * Credit Note
   */
  DelcredereCreditNote = "308",
  /**
   * Proforma invoice
   *
   * Invoice
   */
  ProformaInvoice = "325",
  /**
   * Partial invoice
   *
   * Invoice
   */
  PartialInvoice = "326",
  /**
   * Commercial invoice which includes a packing list
   *
   * Invoice
   */
  CommercialInvoiceWhichIncludesAPackingList = "331",
  /**
   * Commercial invoice
   *
   * Invoice
   */
  CommercialInvoice = "380",
  /**
   * Credit note
   *
   * Credit Note
   */
  CreditNote = "381",
  /**
   * Commission note
   *
   * Invoice
   */
  CommissionNote = "382",
  /**
   * Debit note
   *
   * Invoice
   */
  DebitNote = "383",
  /**
   * Corrected invoice
   *
   * Invoice
   */
  CorrectedInvoice = "384",
  /**
   * Consolidated invoice
   *
   * Invoice
   */
  ConsolidatedInvoice = "385",
  /**
   * Prepayment invoice
   *
   * Invoice
   */
  PrepaymentInvoice = "386",
  /**
   * Hire invoice
   *
   * Invoice
   */
  HireInvoice = "387",
  /**
   * Tax invoice
   *
   * Invoice
   */
  TaxInvoice = "388",
  /**
   * Self-billed invoice
   *
   * Invoice
   */
  SelfBilledInvoice = "389",
  /**
   * Delcredere invoice
   *
   * Invoice
   */
  DelcredereInvoice = "390",
  /**
   * Factored invoice
   *
   * Invoice
   */
  FactoredInvoice = "393",
  /**
   * Lease invoice
   *
   * Invoice
   */
  LeaseInvoice = "394",
  /**
   * Consignment invoice
   *
   * Invoice
   */
  ConsignmentInvoice = "395",
  /**
   * Factored credit note
   *
   * Credit Note
   */
  FactoredCreditNote = "396",
  /**
   * Optical Character Reading (OCR) payment credit note
   *
   * Credit Note
   */
  OpticalCharacterReadingOcrPaymentCreditNote = "420",
  /**
   * Debit advice
   *
   * Invoice
   */
  DebitAdvice = "456",
  /**
   * Reversal of debit
   *
   * Invoice
   */
  ReversalDebit = "457",
  /**
   * Reversal of credit
   *
   * Credit Note
   */
  ReversalCredit = "458",
  /**
   *  Self-billed corrective invoice, invoice type, Corrected
   *
   * Invoice
   */
  SelfBilledCorrectiveInvoiceInvoiceTypeCorrected = "471",
  /**
   *  Factored Corrective Invoice, invoice type, Corrected
   *
   * Invoice
   */
  FactoredCorrectiveInvoiceInvoiceTypeCorrected = "472",
  /**
   *  Self billed Factored corrective invoice, invoice type, Corrected
   *
   * Invoice
   */
  SelfBilledFactoredCorrectiveInvoiceInvoiceTypeCorrected = "473",
  /**
   *  Self Prepayment invoice, invoice type, Original
   *
   * Invoice
   */
  SelfPrepaymentInvoiceInvoiceTypeOriginal = "500",
  /**
   *  Self billed factored invoice, invoice type, Original
   *
   * Invoice
   */
  SelfBilledFactoredInvoiceInvoiceTypeOriginal = "501",
  /**
   *  Self billet factored Credit Note, Credit note type, Corrected
   *
   * Credit Note
   */
  SelfBilletFactoredCreditNoteCreditNoteTypeCorrected = "502",
  /**
   *  Prepayment credit note, credit note type, Corrected
   *
   * Credit Note
   */
  PrepaymentCreditNoteCreditNoteTypeCorrected = "503",
  /**
   * Self billed debit note
   *
   * Invoice
   */
  SelfBilledDebitNote = "527",
  /**
   * Forwarder's credit note
   *
   * Credit Note
   */
  ForwardersCreditNote = "532",
  /**
   * Forwarder's invoice discrepancy report
   *
   * Invoice
   */
  ForwardersInvoiceDiscrepancyReport = "553",
  /**
   * Insurer's invoice
   *
   * Invoice
   */
  InsurersInvoice = "575",
  /**
   * Forwarder's invoice
   *
   * Invoice
   */
  ForwardersInvoice = "623",
  /**
   * Port charges documents
   *
   * Invoice
   */
  PortChargesDocuments = "633",
  /**
   * Invoice information for accounting purposes
   *
   * Invoice
   */
  InvoiceInformationForAccountingPurposes = "751",
  /**
   * Freight invoice
   *
   * Invoice
   */
  FreightInvoice = "780",
  /**
   * Claim notification
   *
   * Invoice
   */
  ClaimNotification = "817",
  /**
   * Consular invoice
   *
   * Invoice
   */
  ConsularInvoice = "870",
  /**
   * Partial construction invoice
   *
   * Invoice
   */
  PartialConstructionInvoice = "875",
  /**
   * Partial final construction invoice
   *
   * Invoice
   */
  PartialFinalConstructionInvoice = "876",
  /**
   * Final construction invoice
   *
   * Invoice
   */
  FinalConstructionInvoice = "877",
  /**
   * Customs invoice
   *
   * Invoice
   */
  CustomsInvoice = "935",
}

export function description(value: Enum1001): string {
  switch (value) {
    case Enum1001.RequestForPayment:
      return "Request for payment";
    case Enum1001.DebitNoteRelatedToGoodsOrServices:
      return "Debit note related to goods or services";
    case Enum1001.CreditNoteRelatedToGoodsOrServices:
      return "Credit note related to goods or services";
    case Enum1001.MeteredServicesInvoice:
      return "Metered services invoice";
    case Enum1001.CreditNoteRelatedToFinancialAdjustments:
      return "Credit note related to financial adjustments";
    case Enum1001.DebitNoteRelatedToFinancialAdjustments:
      return "Debit note related to financial adjustments";
    case Enum1001.TaxNotification:
      return "Tax notification";
    case Enum1001.InvoicingDataSheet:
      return "Invoicing data sheet";
    case Enum1001.DirectPaymentValuation:
      return "Direct payment valuation";
    case Enum1001.ProvisionalPaymentValuation:
      return "Provisional payment valuation";
    case Enum1001.PaymentValuation:
      return "Payment valuation";
    case Enum1001.InterimApplicationForPayment:
      return "Interim application for payment";
    case Enum1001.FinalPaymentRequestBasedOnCompletionWork:
      return "Final payment request based on completion of work";
    case Enum1001.PaymentRequestForCompletedUnits:
      return "Payment request for completed units";
    case Enum1001.SelfBilledCreditNote:
      return "Self billed credit note";
    case Enum1001.ConsolidatedCreditNoteGoodsAndServices:
      return "Consolidated credit note - goods and services";
    case Enum1001.PriceVariationInvoice:
      return "Price variation invoice";
    case Enum1001.CreditNoteForPriceVariation:
      return "Credit note for price variation";
    case Enum1001.DelcredereCreditNote:
      return "Delcredere credit note";
    case Enum1001.ProformaInvoice:
      return "Proforma invoice";
    case Enum1001.PartialInvoice:
      return "Partial invoice";
    case Enum1001.CommercialInvoiceWhichIncludesAPackingList:
      return "Commercial invoice which includes a packing list";
    case Enum1001.CommercialInvoice:
      return "Commercial invoice";
    case Enum1001.CreditNote:
      return "Credit note";
    case Enum1001.CommissionNote:
      return "Commission note";
    case Enum1001.DebitNote:
      return "Debit note";
    case Enum1001.CorrectedInvoice:
      return "Corrected invoice";
    case Enum1001.ConsolidatedInvoice:
      return "Consolidated invoice";
    case Enum1001.PrepaymentInvoice:
      return "Prepayment invoice";
    case Enum1001.HireInvoice:
      return "Hire invoice";
    case Enum1001.TaxInvoice:
      return "Tax invoice";
    case Enum1001.SelfBilledInvoice:
      return "Self-billed invoice";
    case Enum1001.DelcredereInvoice:
      return "Delcredere invoice";
    case Enum1001.FactoredInvoice:
      return "Factored invoice";
    case Enum1001.LeaseInvoice:
      return "Lease invoice";
    case Enum1001.ConsignmentInvoice:
      return "Consignment invoice";
    case Enum1001.FactoredCreditNote:
      return "Factored credit note";
    case Enum1001.OpticalCharacterReadingOcrPaymentCreditNote:
      return "Optical Character Reading (OCR) payment credit note";
    case Enum1001.DebitAdvice:
      return "Debit advice";
    case Enum1001.ReversalDebit:
      return "Reversal of debit";
    case Enum1001.ReversalCredit:
      return "Reversal of credit";
    case Enum1001.SelfBilledCorrectiveInvoiceInvoiceTypeCorrected:
      return " Self-billed corrective invoice, invoice type, Corrected";
    case Enum1001.FactoredCorrectiveInvoiceInvoiceTypeCorrected:
      return " Factored Corrective Invoice, invoice type, Corrected";
    case Enum1001.SelfBilledFactoredCorrectiveInvoiceInvoiceTypeCorrected:
      return " Self billed Factored corrective invoice, invoice type, Corrected";
    case Enum1001.SelfPrepaymentInvoiceInvoiceTypeOriginal:
      return " Self Prepayment invoice, invoice type, Original";
    case Enum1001.SelfBilledFactoredInvoiceInvoiceTypeOriginal:
      return " Self billed factored invoice, invoice type, Original";
    case Enum1001.SelfBilletFactoredCreditNoteCreditNoteTypeCorrected:
      return " Self billet factored Credit Note, Credit note type, Corrected";
    case Enum1001.PrepaymentCreditNoteCreditNoteTypeCorrected:
      return " Prepayment credit note, credit note type, Corrected";
    case Enum1001.SelfBilledDebitNote:
      return "Self billed debit note";
    case Enum1001.ForwardersCreditNote:
      return "Forwarder's credit note";
    case Enum1001.ForwardersInvoiceDiscrepancyReport:
      return "Forwarder's invoice discrepancy report";
    case Enum1001.InsurersInvoice:
      return "Insurer's invoice";
    case Enum1001.ForwardersInvoice:
      return "Forwarder's invoice";
    case Enum1001.PortChargesDocuments:
      return "Port charges documents";
    case Enum1001.InvoiceInformationForAccountingPurposes:
      return "Invoice information for accounting purposes";
    case Enum1001.FreightInvoice:
      return "Freight invoice";
    case Enum1001.ClaimNotification:
      return "Claim notification";
    case Enum1001.ConsularInvoice:
      return "Consular invoice";
    case Enum1001.PartialConstructionInvoice:
      return "Partial construction invoice";
    case Enum1001.PartialFinalConstructionInvoice:
      return "Partial final construction invoice";
    case Enum1001.FinalConstructionInvoice:
      return "Final construction invoice";
    case Enum1001.CustomsInvoice:
      return "Customs invoice";
  }
}
